src/ngraph/node.hpp

```diff
@@ -139,6 +148,41 @@ namespace ngraph
     public:
         virtual ~Node();
 
+        /// Tests if a node is of op type T
+        template <typename NodeType>
+        bool has_type() const
+        {
+            return &get_type_info() == &NodeType::type_info;
+        }
+
+        /// Casts a Node to a shared_ptr<T> if is of type T, nullptr otherwise;
+        template <typename NodeType>
+        std::shared_ptr<NodeType> as_type()
+        {
+            return has_type<NodeType>() ? shared_from_this() : std::shared_ptr<NodeType>();
+        }
+
+        /// Casts a Node to a shared_ptr<T> if is of type T, nullptr otherwise;
+        template <typename NodeType>
+        std::shared_ptr<const NodeType> as_type() const
+        {
+            return has_type<NodeType>() ? shared_from_this() : std::shared_ptr<NodeType>();
+        }
+
+        /// Returns the NodeTypeInfo for the node's class.
+        /// During transition to type_info, return's a dummy type_info for Node if the class
+        /// has not been updated yet.
+        virtual const NodeTypeInfo& get_type_info() const { return type_info; }
+        virtual const char* get_type_name() const
```

Comment:
This should return a string to be more useful
With this I can't say `if (node.get_type_name() == "Abs")`

Reply:
You shouldn't be doing that anyways; there can be more than one version of Abs. It's also an accessor; `description` returns a string.